uaclient/cli.py

```diff
@@ -117,9 +117,15 @@ def new_f(args, cfg, **kwargs):
             notice_msg = "Operation in progress: {}".format(lock_holder)
             cfg.add_notice("", notice_msg)
             _CLEAR_LOCK_FILE = cfg.delete_cache_key
-            retval = f(args, cfg, **kwargs)
-            cfg.delete_cache_key("lock")
-            _CLEAR_LOCK_FILE = None  # Unset due to successful lock delete
+
+            try:
+                retval = f(*args, cfg=cfg, **kwargs)
+            except Exception as e:
+                raise e
+            finally:
+                cfg.delete_cache_key("lock")
+                _CLEAR_LOCK_FILE = None  # Unset due to successful lock delete
```

Comment:
Not really a blocker, but I don't think we need the except/raise here either, the exception will be raised even without just the try/finally.

```suggestion
            finally:
                cfg.delete_cache_key("lock")
                _CLEAR_LOCK_FILE = None  # Unset due to successful lock delete
```